presentation/src/main/java/com/binhth/photocollection/ui/screen/collection/ListCollectionFragment.kt

```diff
@@ -0,0 +1,91 @@
+package com.binhth.photocollection.ui.screen.collection
+
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+import com.binhth.photocollection.BR
+import com.binhth.photocollection.databinding.FragmentListItemBinding
+import com.binhth.photocollection.model.CollectionItem
+import com.binhth.photocollection.ui.screen.core.BaseListFragment
+import com.binhth.photocollection.utils.EndlessScrollListener
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import java.util.*
+
+
+class ListCollectionFragment : BaseListFragment<FragmentListItemBinding, ListCollectionViewModel, CollectionItem>(),
+    SwipeRefreshLayout.OnRefreshListener {
+    companion object {
+        const val TAG = "ListCollectionFragment"
+
+        fun newInstance() = ListCollectionFragment()
+    }
+
+    override val bindingVariable: Int
+        get() = BR.viewModel
+
+    override val viewModel by viewModel<ListCollectionViewModel>()
+
+    private lateinit var endlessScrollListener: EndlessScrollListener
+
+    override fun initContent(viewBinding: FragmentListItemBinding) {
+        viewBinding.viewModel = viewModel
+        viewBinding.swipeRefresh.setOnRefreshListener(this@ListCollectionFragment)
+        viewModel.apply {
+            val listCollectionAdapter = ListCollectionAdapter(itemClick = { TODO("open photo list") })
+            val gridLayoutManager = GridLayoutManager(context, 2)
+
+            endlessScrollListener = EndlessScrollListener(onLoadMore = {
+                if (isLoading.value == true
+                    || isRefresh.value == true
+                    || isLoadMore.value == true
+                ) else {
+                    isLoadMore.value = true
+                    viewModel.loadMore()
+                }
+            })
+            viewBinding.recyclerView.apply {
+                layoutManager = gridLayoutManager
+                adapter = listCollectionAdapter
+                addOnScrollListener(endlessScrollListener)
+            }
+
+            initLoad()
+
+            isRefresh.observe(this@ListCollectionFragment, Observer {
+                viewBinding.swipeRefresh.isRefreshing = it == true
+            })
+
+            listItem.observe(this@ListCollectionFragment, Observer {
+                val list = ArrayList<CollectionItem>()
+                list.addAll(it)
```

Comment:
A làm thế này liệu có làm được load more không nhỉ?

Reply:
Đoạn này a test loadmore thấy ok k vấn đề

Reply:
chỗ này hơi lạ nhỉ, vì khác gì lần nào cũng chỉ submitList với cái list mới get về

Reply:
trong  onSuccess() của BaseListViewModel.kt mình có gán lại giá trị cho listitem này với giá trị cũ + thêm giá trị từ load more rồi

Reply:
à ok mình nhầm, cơ mà thế thì submitList(it) luôn đc ko nhỉ,?